mempool/mempool.go

```diff
@@ -187,8 +187,14 @@ func (mem *Mempool) CheckTx(tx types.Tx, cb func(*abci.Response)) (err error) {
 	// WAL
 	if mem.wal != nil {
 		// TODO: Notify administrators when WAL fails
-		mem.wal.Write([]byte(tx))
-		mem.wal.Write([]byte("\n"))
+		_, err := mem.wal.Write([]byte(tx))
```

Comment:
Not sure if we should return right away. The mempool wal is a more optional feature that shouldnt inhibit the mempool. We should log the error and leave the `TODO: Notify administrators `...